amqp/src/test/java/io/github/tcdl/adapters/amqp/AmqpConnectionManagerTest.java

```diff
@@ -0,0 +1,100 @@
+package io.github.tcdl.adapters.amqp;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+import static org.mockito.Mockito.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import io.github.tcdl.config.amqp.AmqpBrokerConfig;
+
+import java.io.IOException;
+import java.util.Optional;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import com.rabbitmq.client.Connection;
+import com.rabbitmq.client.ConnectionFactory;
+
+public class AmqpConnectionManagerTest {
+    
+    final String host = "127.0.0.1";
+    final int port = 5672;
+    final String username = "user";
+    final String password = "pwd";
+    final String virtualHost = "127.10.10.10"; 
+    final String groupId = "msb-java";
+    final boolean durable = Boolean.valueOf(false);
+    final int consumerThreadPoolSize = 5;
+    
+    ConnectionFactory mockConnectionFactory;
+    Connection mockConnection;
+    AmqpBrokerConfig amqpConfig;
+    AmqpConnectionManager amqpConnectionManager;
+    
+    @Before
+    public void setUp() {
+        mockConnectionFactory = mock(ConnectionFactory.class);
+        mockConnection = mock(Connection.class);
+        
+        amqpConfig = new AmqpBrokerConfig(host, port, 
+                Optional.of(username), Optional.of(password), Optional.of(virtualHost), groupId, durable, consumerThreadPoolSize);
+        
+        try {
+            when(mockConnectionFactory.newConnection()).thenReturn(mockConnection);
+        } catch (IOException e) {
+            fail("Can't create MockConnectionFactory");
+        }
+
+        amqpConnectionManager = new AmqpConnectionManager(amqpConfig) {
```

Comment:
Can we refactor AmqpConnectionManager to make connection factory injected and avoid modification of the class that is under test?